cppclient: fix warnings

```
thibault@MacBook-Pro-de-Thibault ~/i/entangled> bazel build -c opt //cppclient/...
INFO: Analyzed 4 targets (34 packages loaded, 6286 targets configured).
INFO: Found 4 targets...
INFO: From Compiling cppclient/api_json.cc:
cppclient/api_json.cc:485:54: warning: missing field 'error' initializer [-Wmissing-field-initializers]
  return {response["states"].get<std::vector<bool>>()};
                                                     ^
cppclient/api_json.cc:506:54: warning: missing field 'error' initializer [-Wmissing-field-initializers]
  return {response["states"].get<std::vector<bool>>()};
                                                     ^
2 warnings generated.
INFO: Elapsed time: 20.244s, Critical Path: 16.72s
INFO: 111 processes: 111 darwin-sandbox.
INFO: Build completed successfully, 181 total actions
thibault@MacBook-Pro-de-Thibault ~/i/entangled> 
```

--- a/cppclient/api.h
+++ b/cppclient/api.h
@@ -91,12 +91,13 @@ class IotaAPI {
 
   virtual bool storeTransactions(const std::vector<std::string>& trytes) = 0;
 
-  virtual GetInclusionStatesResponse getInclusionStates(const std::vector<std::string>& trans,
-                                                        const std::vector<std::string>& tips) = 0;
+  virtual nonstd::optional<GetInclusionStatesResponse> getInclusionStates(const std::vector<std::string>& trans,
+                                                                          const std::vector<std::string>& tips) = 0;
 
   virtual bool broadcastTransactions(const std::vector<std::string>& trytes) = 0;
 
-  virtual WereAddressesSpentFromResponse wereAddressesSpentFrom(const std::vector<std::string>& addresses) = 0;
+  virtual nonstd::optional<WereAddressesSpentFromResponse> wereAddressesSpentFrom(
+      const std::vector<std::string>& addresses) = 0;
 };
 }  // namespace cppclient
 

--- a/cppclient/api_json.cc
+++ b/cppclient/api_json.cc
@@ -462,8 +462,8 @@ std::unordered_set<std::string> IotaJsonAPI::filterConsistentTails(const std::ve
   return ret;
 }
 
-GetInclusionStatesResponse IotaJsonAPI::getInclusionStates(const std::vector<std::string>& trans,
-                                                           const std::vector<std::string>& tips) {
+nonstd::optional<GetInclusionStatesResponse> IotaJsonAPI::getInclusionStates(const std::vector<std::string>& trans,
+                                                                             const std::vector<std::string>& tips) {
   VLOG(3) << __FUNCTION__;
   json req;
   req["command"] = "getInclusionStates";
@@ -482,10 +482,11 @@ GetInclusionStatesResponse IotaJsonAPI::getInclusionStates(const std::vector<std
     return {};
   }
 
-  return {response["states"].get<std::vector<bool>>()};
+  return nonstd::optional<GetInclusionStatesResponse>({response["states"].get<std::vector<bool>>()});
 }
 
-WereAddressesSpentFromResponse IotaJsonAPI::wereAddressesSpentFrom(const std::vector<std::string>& addresses) {
+nonstd::optional<WereAddressesSpentFromResponse> IotaJsonAPI::wereAddressesSpentFrom(
+    const std::vector<std::string>& addresses) {
   VLOG(3) << __FUNCTION__;
   json req;
   req["command"] = "wereAddressesSpentFrom";
@@ -503,7 +504,7 @@ WereAddressesSpentFromResponse IotaJsonAPI::wereAddressesSpentFrom(const std::ve
     return {};
   }
 
-  return {response["states"].get<std::vector<bool>>()};
+  return nonstd::optional<WereAddressesSpentFromResponse>({response["states"].get<std::vector<bool>>()});
 }
 
 template <typename T>

--- a/cppclient/api_json.h
+++ b/cppclient/api_json.h
@@ -56,10 +56,11 @@ class IotaJsonAPI : virtual public IotaAPI {
   bool storeTransactions(const std::vector<std::string>& trytes) override;
   bool broadcastTransactions(const std::vector<std::string>& trytes) override;
 
-  GetInclusionStatesResponse getInclusionStates(const std::vector<std::string>& trans,
-                                                const std::vector<std::string>& tips) override;
+  nonstd::optional<GetInclusionStatesResponse> getInclusionStates(const std::vector<std::string>& trans,
+                                                                  const std::vector<std::string>& tips) override;
 
-  WereAddressesSpentFromResponse wereAddressesSpentFrom(const std::vector<std::string>& addresses) override;
+  nonstd::optional<WereAddressesSpentFromResponse> wereAddressesSpentFrom(
+      const std::vector<std::string>& addresses) override;
 
   virtual nonstd::optional<nlohmann::json> post(const nlohmann::json& input) = 0;
 

--- a/cppclient/messages.h
+++ b/cppclient/messages.h
@@ -44,12 +44,10 @@ struct GetNodeInfoResponse {
 
 struct GetInclusionStatesResponse {
   std::vector<bool> states;
-  nonstd::optional<std::string> error;
 };
 
 struct WereAddressesSpentFromResponse {
   std::vector<bool> states;
-  nonstd::optional<std::string> error;
 };
 }  // namespace cppclient
 

--- a/tanglescope/README.md
+++ b/tanglescope/README.md
@@ -28,18 +28,18 @@ Measuring transactions/bundles traffic data by listening to a list of zmq publis
 ```yaml
 statscollector:
 
-  #IP/Port that the Prometheus Exposer binds to
+#IP / Port that the Prometheus Exposer binds to
   prometheus_exposer_uri: "0.0.0.0:8081"
-  #URL of ZMQ publisher to connect to
+#URL of ZMQ publisher to connect to
   publishers:
       - "tcp://zmq.testnet.iota.org:5556"
 
-  #histogram of bundle's confirmation duration granularity:
-  #(number of buckets = bundle_confirmation_histogram_range*1000 / bundle_confirmation_bucket_size)
+#histogram of bundle's confirmation duration granularity:
+#(number of buckets = bundle_confirmation_histogram_range * 1000 / bundle_confirmation_bucket_size)
 
-  #histogram's range  [seconds]
+#histogram's range  [seconds]
   bundle_confirmation_histogram_range: 240
-  #bucket's size [seconds]
+#bucket's size [seconds]
   bundle_confirmation_bucket_size: 2
 ```
 
@@ -84,22 +84,22 @@ Brief description of the logic:
 
 ```yaml
   echocollector:
-  #list of publishers to listen to. (they should return the echo)
+#list of publishers to listen to.(they should return the echo)
   publishers:
       - "tcp://zmq.testnet.iota.org:5556"
 
-  #url of iri node
+#url of iri node
   iri_host: "iri01.testnet.iota.cafe"
   iri_port: 14265
-  #IP/Port that the Prometheus Exposer binds to
+#IP / Port that the Prometheus Exposer binds to
   prometheus_exposer_uri: "0.0.0.0:8080"
-  #MWM for transaction's POW
+#MWM for transaction's POW
   mwm: 9
-  #in-between transactions interval [seconds]
+#in - between transactions interval[seconds]
   broadcast_interval: 20
-  #how often should echocatcher discover "unseen" transactions [seconds]
-  #shouldn't be too frequent since it can incur in multiple api
-  #calls ..
+#how often should echocatcher discover "unseen" transactions[seconds]
+#shouldn't be too frequent since it can incur in multiple api
+#calls..
   discovery_interval: 5  
 ```
 
@@ -108,11 +108,11 @@ Also requires:
 ```yaml
 dbloader:
 
-  #URL of ZMQ publisher to connect to
+#URL of ZMQ publisher to connect to
   publisher: "tcp://zmq.testnet.iota.org:5556"
-  #TX's older than that age will be removed each cleanup_interval [seconds]
+#TX's older than that age will be removed each cleanup_interval [seconds]
   oldest_tx_age: 1800
-  #in - between cleanup interval[seconds]
+#in - between cleanup interval[seconds]
   cleanup_interval: 60
 ```
 
@@ -137,17 +137,17 @@ and ref counting transaction's direct approvers
 ```yaml
 blowballcollector:
 
-  #IP/Port that the Prometheus Exposer binds to
+#IP / Port that the Prometheus Exposer binds to
   prometheus_exposer_uri: "0.0.0.0:8082"
-  #URL of ZMQ publisher to connect to
+#URL of ZMQ publisher to connect to
   publisher: "tcp://zmq.testnet.iota.org:5556"
-  #interval between following snapshots (internal counters update)
+#interval between following snapshots(internal counters update)
   snapshot_interval: 60
 
-  #histogram granularity configuration
-  #histogram's bucket's size (refcount of tx approvers)
+#histogram granularity configuration
+#histogram 's bucket' s size(refcount of tx approvers)
   bucket_size: 1
-  #histogram's range (maximum value of refcount to report)
+#histogram's range (maximum value of refcount to report)
   histogram_range: 40up_interval: 60
 ```
 
@@ -166,21 +166,21 @@ Monitoring tip selection's algorithm behavior.
 ```yaml
 tipselectioncollector:
 
-  #IP / Port that the Prometheus Exposer binds to
+#IP / Port that the Prometheus Exposer binds to
   prometheus_exposer_uri: "0.0.0.0:8083"
-  #url of iri node
+#url of iri node
   iri_host : "iri01.testnet.iota.cafe"
   iri_port: 14265
-  #depth to give to "getTransactionsToApprove"
+#depth to give to "getTransactionsToApprove"
   depth: 10
-  #in - between samples interval[seconds]
+#in - between samples interval[seconds]
   sample_interval: 20
-  #of calls to getTransactionsToApprove in each sample
+#of calls to getTransactionsToApprove in each sample
   sample_size: 20
-  #histogram granularity configuration
-  #duration's histogram (time it took 'getTransactionsToApprove') bucket's size [ms]
+#histogram granularity configuration
+#duration 's histogram (time it took ' getTransactionsToApprove ') bucket' s size[ms]
   duration_bucket_size: 20
-  #duration's histogram range (maximum time it took 'getTransactionsToApprove')
+#duration 's histogram range (maximum time it took ' getTransactionsToApprove')
   duration_histogram_range: 1000
 ```
 
@@ -214,15 +214,15 @@ Measuring tangle width as defined below:
 ```yaml
 tanglewidthcollector:
 
-  #IP / Port that the Prometheus Exposer binds to
+#IP / Port that the Prometheus Exposer binds to
   prometheus_exposer_uri: "0.0.0.0:8084"
-  #The measure line base from the above definition, [seconds]
+#The measure line base from the above definition, [seconds]
   measure_line_base_age: 30
-  #The measure line max age [seconds]
+#The measure line max age[seconds]
   measure_line_max_age: 90
-  #Step from measure_line_base_age to measure_line_max_age, [seconds]
+#Step from measure_line_base_age to measure_line_max_age, [seconds]
   measure_line_age_step: 10
-  #interval between following snapshots (internal counters update)
+#interval between following snapshots(internal counters update)
   snapshot_interval: 10
 ```
 
@@ -231,11 +231,11 @@ Also requires:
 ```yaml
 dbloader:
 
-  #URL of ZMQ publisher to connect to
+#URL of ZMQ publisher to connect to
   publisher: "tcp://zmq.testnet.iota.org:5556"
-  #TX's older than that age will be removed each cleanup_interval [seconds]
+#TX's older than that age will be removed each cleanup_interval [seconds]
   oldest_tx_age: 1800
-  #in - between cleanup interval[seconds]
+#in - between cleanup interval[seconds]
   cleanup_interval: 60
 ```
 
@@ -258,30 +258,30 @@ Measuring transaction's confirmation rate
 
 ```yaml
 confirmationratecollector:
-  #list of publisher to listen to. (they should return the echo)
+#list of publisher to listen to.(they should return the echo)
   publishers:
       - "tcp://zmq.testnet.iota.org:5556"
 
-  #url of iri node
+#url of iri node
   iri_host: "iri01.testnet.iota.cafe"
   iri_port: 14265
-  #IP/Port that the Prometheus Exposer binds to
+#IP / Port that the Prometheus Exposer binds to
   prometheus_exposer_uri: "0.0.0.0:8085"
-  #MWM for transaction's POW
+#MWM for transaction's POW
   mwm: 9
-  #in-between transactions interval [seconds]
+#in - between transactions interval[seconds]
   broadcast_interval: 3
 
-  #Time after which we should expect transaction to be approved
+#Time after which we should expect transaction to be approved
   measurement_upper_bound: 60
-  #Time after which if a transaction hasn't been approved, we give up
+#Time after which if a transaction hasn't been approved, we give up
   measurement_lower_bound: 360
-  #should calculate confirmation rate using API calls
+#should calculate confirmation rate using API calls
   enable_cr_from_api: true
-  #For checking the CR for various Latencies, define the latency in seconds,
-  #"step" means that for 10 seconds we will check latency of 10,20,30 seconds
+#For checking the CR for various Latencies, define the latency in seconds,
+#"step" means that for 10 seconds we will check latency of 10, 20, 30 seconds
   additional_latency_step_seconds: 10
-  #num different latencies to check
+#num different latencies to check
   additional_latency_num_steps: 3
 ```
 
@@ -290,6 +290,3 @@ confirmationratecollector:
 - confirmationratecollector_confirmation_rate_api (Confirmation rate ratio [0,1] as it is perceived by making api calls to "getInclusionStates") [Gauge]  
 
 - confirmationratecollector_confirmation_rate_zmq (Confirmation rate ratio [0,1] as it is perceived by inspecting the zmq stream) [Gauge]
-
-
-

--- a/tanglescope/common/txauxiliary.cpp
+++ b/tanglescope/common/txauxiliary.cpp
@@ -34,12 +34,12 @@ boost::future<void> removeConfirmedTransactions(std::weak_ptr<cppclient::IotaAPI
                                                tips] { return clientSharedPtr->getInclusionStates(txs, tips); })
         .then([&](auto future) {
           auto response = future.get();
-          if (response.states.empty()) {
+          if (!response.has_value()) {
             txs.clear();
             return;
           }
 
-          auto states = std::move(response.states);
+          auto states = std::move(response.value().states);
           auto idx = 0;
           txs.erase(std::remove_if(txs.begin(), txs.end(), [&idx, &states](std::string hash) { return states[idx++]; }),
                     txs.end());

--- a/tanglescope/confirmationratecollector.cpp
+++ b/tanglescope/confirmationratecollector.cpp
@@ -93,14 +93,14 @@ void CRCollector::calcConfirmationRateAPICall() {
   }
 
   auto resp = _api->getInclusionStates(transactions, tips);
-  if (resp.error.has_value() || resp.states.size() != transactions.size()) {
+  if (!resp.has_value() || resp.value().states.size() != transactions.size()) {
     return;
   }
   std::set<std::string> confirmedTransactions;
   uint32_t idx = 0;
   std::copy_if(transactions.begin(), transactions.end(),
                std::inserter(confirmedTransactions, confirmedTransactions.begin()),
-               [&](const std::string& hash) { return resp.states[idx++]; });
+               [&](const std::string& hash) { return resp.value().states[idx++]; });
 
   calcAndExposeImpl(confirmedTransactions, CONFIRMATION_RATE_API);
 }

--- a/tanglescope/doxy.txt
+++ b/tanglescope/doxy.txt
@@ -5,7 +5,7 @@
  * Refer to the LICENSE file for licensing information
  */
 
- /**
+/**
  * @defgroup tanglescope Tanglescope
  * @brief Tanglescope is a program composed of metric collectors.
  *

